zowe-rest-api-sample-spring/docs/api-client-authentication.md

```diff
@@ -0,0 +1,201 @@
+    Authorization: Basic em93ZTp6b3dl
+    ```
+
+2. If the credentials are valid then the server responds with status code based on the success of the endpoint itself:
+
+    ```http
+    HTTP/1.1 200
+    {
+        ...
+    }
+    ```
+
+### Unauthenticated request error
+
+If the authentication fails from any reason except for internal server error then [`401`](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/401) status is returned. This status is sent with a `WWW-Authenticate` header that contains information on how to authorize correctly and contains the name of the service.
+
+```http
+GET /api/v1/greeting HTTP/1.1
+
+HTTP/1.1 401
+WWW-Authenticate: Basic realm="Zowe Sample API Service", charset="UTF-8"
+```
+
+The Zowe REST API SDK provides more details in the JSON response.
+This is optional and is meant for diagnostics purposes.
+
+### The support for HTTP basic authentication in the Zowe REST API SDK
+
+REST API services that are developed by the Zowe REST API SDK support HTTP basic authentication by default. When it is running on z/OS then it is using the SAF interface to validate the user ID and password by the z/OS security subsystems. For testing outside of z/OS, it uses a predefined dummy credentials. For more information see [Web Security](web-security.md) and [z/OS Security](zos-security.md).
+
+## Token-based Authentication
+
+Since every REST request requires to provide valid authentication and REST API services do not have mechanism of a long-term web session, it is necessary to have a way how to establish an authenticated session outside of individual REST API services.
+
+In Zowe and in many other applications, this is achieved using JWT tokens that can be obtained by a specialized serviced and then used as the authentication information. This service is described in more details at [Zowe Authentication and Authorization Service](https://github.com/zowe/api-layer/wiki/Zowe-Authentication-and-Authorization-Service).
+
+This section focuses on the way how services in the Zowe API ecosystem are expected to accept and use these tokens so the API clients have unified experience.
+
+### Token-based Login Flow and Request/Response Format
+
+1. The API client obtains JWT token by using POST method on `/auth/login` endpoint of API service that requires valid user ID and password
+
+2. The API service calls an *authentication provider* that returns a JWT token that contains the user ID claim in the HTTP cookie named `apimlAuthenticationToken` with attributes `HttpOnly` and `Secure`.
+
+3. The API client remembers the JWT token or cookie and sends it with every request as the cookie with name `apimlAuthenticationToken`.
+
+#### Obtaining the token
+
+- The full URL is the base URL of the API service plus `/auth/login`. If the application has the base URL with `/api/v1` then the full URL can be: `https://hostname:port/api/v1/auth/login`.
+
+- The credentials are provide in JSON request:
+
+    ```json
+    {
```

Comment:
Why this over basic auth header?

Reply:
There is nothing bad with Basic auth header and I would prefer it too. I tried to keep it same as the ZAAS and ESM microservice.

Reply:
Gotcha, makes sense, we'll need to make this update in imperative rest client